client/ayon_core/pipeline/farm/pyblish_functions.py

```diff
@@ -515,6 +517,132 @@ def create_instances_for_aov(instance, skeleton, aov_filter,
     )
 
 
+def _get_legacy_product_name_and_group(
+        product_type,
+        source_product_name,
+        task_name,
+        dynamic_data):
+    """Get product name with legacy logic.
+
+    This function holds legacy behaviour of creating product name
+    that is deprecated. This wasn't using product name templates
+    at all, only hardcoded values. It shouldn't be used anymore,
+    but transition to templates need careful checking of the project
+    and studio settings.
+
+    Deprecated:
+        since 0.4.4
+
+    Args:
+        product_type (str): Product type.
+        source_product_name (str): Source product name.
+        task_name (str): Task name.
+        dynamic_data (dict): Dynamic data (camera, aov, ...)
+
+    Returns:
+        tuple: product name and group name
+
+    """
+    warnings.warn("Using legacy product name for renders",
+                  DeprecationWarning)
+
+    if not source_product_name.startswith(product_type):
+        resulting_group_name = '{}{}{}{}{}'.format(
+            product_type,
+            task_name[0].upper(), task_name[1:],
+            source_product_name[0].upper(), source_product_name[1:])
+    else:
+        resulting_group_name = source_product_name
+
+    # create product name `<product type><Task><Product name>`
+    if not source_product_name.startswith(product_type):
+        resulting_group_name = '{}{}{}{}{}'.format(
+            product_type,
+            task_name[0].upper(), task_name[1:],
+            source_product_name[0].upper(), source_product_name[1:])
+    else:
+        resulting_group_name = source_product_name
+
+    resulting_product_name = '{}'.format(resulting_group_name)
+    camera = dynamic_data.get("camera")
+    aov = dynamic_data.get("aov")
+    if camera:
+        if not aov:
+            resulting_product_name = '{}_{}'.format(
+                resulting_group_name, camera)
+        elif not aov.startswith(camera):
+            resulting_product_name = '{}_{}_{}'.format(
+                resulting_group_name, camera, aov)
+        else:
+            resulting_product_name = "{}_{}".format(
+                resulting_group_name, aov)
+    else:
+        if aov:
+            resulting_product_name = '{}_{}'.format(
+                resulting_group_name, aov)
+
+    return resulting_product_name, resulting_group_name
+
+
+def get_product_name_and_group_from_template(
+        task_entity,
+        project_name,
+        host_name,
+        product_type,
+        variant,
```

Comment:
It should probably match the arguments order from `get_product_name`.
```suggestion
        project_name,
        task_entity,
        product_type,
        variant,
        host_name,
```